### PR TITLE
SWARM-637: First attempt at some JAX-WS client tests

### PR DESCRIPTION
**Pull request by sfcoy**
Not working!

### CODE REVIEW

File: testsuite/testsuite-webservices/pom.xml [hunk 1]
```diff
@@ -20,13 +20,34 @@
   <name>Test Suite: WebServices</name>
   <description>Test Suite: WebServices</description>
 
+  <build>
+    <plugins>
+      <plugin>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.19.1</version>
+        <configuration>
+          <argLine>-Xbootclasspath/p:${user.home}/.m2/repository/org/jboss/logmanager/jboss-logmanager/2.0.3.Final/jboss-logmanager-2.0.3.Final.jar
```
Comment (heiko-braun) [inline testsuite/testsuite-webservices/pom.xml]: Why did you add this?

File: testsuite/testsuite-webservices/pom.xml [hunk 2]
```diff
@@ -20,13 +20,34 @@
...
+      <artifactId>jaxrs-cdi</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.wildfly.swarm</groupId>
       <artifactId>webservices</artifactId>
     </dependency>
     <dependency>
       <groupId>org.wildfly.swarm</groupId>
+      <artifactId>container</artifactId>
```
Comment (heiko-braun) [inline testsuite/testsuite-webservices/pom.xml]: this is transitive to all fractions and should not be needed

File: testsuite/testsuite-webservices/pom.xml (hunk 1 above)
Comment (sfcoy) [inline testsuite/testsuite-webservices/pom.xml]: Because depending on how I set things up, I would get complaints from jboss-logging. It seems redundant right now though

File: testsuite/testsuite-webservices/src/test/java/org/wildfly/swarm/webservices/JAXWSClientTest.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    public static Archive createDeployment() {
+        System.out.println("Creating deployment");
+        final WARArchive warArchive = ShrinkWrap.create(WARArchive.class, "myapp.war")
+                .addClass(WebserviceClientServlet.class)
+                .addClass(WebserviceClientWithHandlerServlet.class)
+                .addClass(SampleSoapHandler.class)
+                .addClass(SimpleWSServer.class)
+                .addClass(SimpleWebserviceEndpointImpl.class)
+                .addClass(SimpleWebserviceEndpointIface.class);
+        warArchive.addModule("javax.xml.ws.api");
```
Comment (sfcoy) [inline testsuite/testsuite-webservices/src/test/java/org/wildfly/swarm/webservices/JAXWSClientTest.java]: This `addModule` was frankly a wild guess....

File: testsuite/testsuite-webservices/src/test/java/org/wildfly/swarm/webservices/JAXWSClientTest.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+    public Container newContainer(String... args) throws Exception {
+        wsServer = new SimpleWSServer();
+        wsServer.buildAndStartWebService();
+        System.out.println("Building container");
+        return new Container(false)
+                .fraction(WebServicesFraction.createDefaultFraction())
+                .fraction(buildNamingFraction());
+    }
+
```
Comment (sfcoy) [inline testsuite/testsuite-webservices/src/test/java/org/wildfly/swarm/webservices/JAXWSClientTest.java]: Ignore the two tests below, they require the presence of an @Before, @After methods that set up and take down the web service. They just prove that the basic machinery works out of container

**Comment by bobmcwhirter** [comment]
test this please

**Comment by sfcoy** [comment]
This is not intended to be merged just yet; it was provided at Heiko's request

**Comment by heiko-braun** [comment]
@sfcoy I just returned from vacation and lost the context somehow. What did we discuss on how to proceed with this?

**Comment by heiko-braun** [comment]
IIRC we had some discussion about the purpose of `SimpleWSServer`. I think that would not be needed when the endpoints are deployed to Swarm.

**Comment by sfcoy** [comment]
@heiko-braun I have not had the time to follow it up myself. Ultimately I'm trying to write some tests that prove that there is a problem when adding javax.xml.ws.handler.Handlers into a JAX-WS client.
 It's related to [Web service InvocationTargetException when using a custom javax.xml.ws.handler.Handler](https://groups.google.com/forum/#!topic/wildfly-swarm/NHMrt1pFpWA)

**Comment by sfcoy** [comment]
Therefore I'm trying to bootstrap a simple WS server for the client test to call.

**Comment by heiko-braun** [comment]
Thanks the link to the forum post is helpful.

**Comment by heiko-braun** [comment]
@sfcoy Thanks I've captured the summary here: https://issues.jboss.org/browse/SWARM-637 and scheduled it for review in the next sprint (2016.10)

**Comment by sfcoy** [comment]
This test is not yet working. We need a way to set up a WS stub or server for the client to call.

**Comment by gastaldi** [comment]
I think https://github.com/wildfly-swarm/wildfly-swarm/pull/121 might fix this problem. Can you give it a try?

**Comment by kenfinnigan** [comment]
Closing in favor of https://github.com/wildfly-swarm/wildfly-swarm/pull/162 that adds test and fixes problem
